dialog/base.go

```diff
@@ -140,6 +140,11 @@ func (d *dialog) setButtons(buttons fyne.CanvasObject) {
 	d.win.Refresh()
 }
 
+func (d *dialog) setIcon(icon fyne.Resource) {
+	d.win.Content.(*fyne.Container).Objects[0] = &canvas.Image{Resource: icon}
```

Comment:
Please see the discussion in https://github.com/fyne-io/fyne/pull/4615. Would you mind making sure that the object is set to a spacer in case of no icon (icon == nil). This reduces CPU usage of trying to draw an empty image.